File: 0062-unique-paths/0062-unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        if (m==n && m==1)
        {
            return 1;
        }
        vector<int> vec;
        vector<vector<int>> dp;
        for(int i=0;i<m;i++)
         {   for(int j=0;j<n;j++)
            vec.push_back(1);
            dp.push_back(vec);
         }
        
        for(int i=1;i<m;i++)
        {
            for(int j=1;j<n;j++)
            {
                dp[i][j] = dp[i-1][j] + dp[i][j-1];
            }
        }

        return dp[m-1][n-1];
        
    }
};

File: 0062-unique-paths/0062-unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        #recursive soln is exhaustive 
        if m==n==1:
            return 1 
        dp = [[0]*n]*m
        for i in range(m):
            dp[i][0]=1
        for j in range(n):
            dp[0][j]=1
        for i in range(1,m):
            for j in range(1,n):
                dp[i][j] = dp[i-1][j]+dp[i][j-1]
        return dp[-1][-1]
        